reader.go

```diff
@@ -80,7 +80,8 @@ func (r *Reader) ReadAt(p []byte, off int64) (int, error) {
 	}
 
 	if off+int64(l) > r.Length() {
-		return 0, errors.New("read beyond end of file")
+		l = int(r.Length() - off)
+		p = p[:l]
```

Comment:
I believe this is handled by 123, below, but the unit test I've added is far from exhaustive. What are your thoughts here? The docs are a little vague.